packages/bridge/src/runtime/head/plugins/unhead.ts

```diff
@@ -0,0 +1,43 @@
+import { createHead as createClientHead, createServerHead, Vue2ProvideUnheadPlugin } from '@unhead/vue'
+import { markRaw } from 'vue'
+import { renderSSRHead } from '@unhead/ssr'
+import { defineNuxtPlugin } from '../../app'
+// @ts-ignore
+import metaConfig from '#build/meta.config.mjs'
+
+export default defineNuxtPlugin((nuxtApp) => {
+  const createHead = process.server ? createServerHead : createClientHead
+  const head = createHead()
+  head.push(markRaw(metaConfig.globalMeta))
+
+  nuxtApp.vueApp.use(Vue2ProvideUnheadPlugin, head)
+  nuxtApp.vueApp.use(head)
+
+  if (process.client) {
+    // pause dom updates until page is ready and between page transitions
+    let pauseDOMUpdates = true
+    const unpauseDom = () => {
+      pauseDOMUpdates = false
+      // trigger the debounced DOM update
+      head.hooks.callHook('entries:updated', head)
+    }
+
+    head.hooks.hook('dom:beforeRender', (context) => { context.shouldRender = !pauseDOMUpdates })
+
+    nuxtApp.hooks.hook('render:before', () => { pauseDOMUpdates = true })
+    // wait for new page before unpausing dom updates
+    nuxtApp.hooks.hook('render:done', unpauseDom)
```

Comment:
`render:done` is a server-side hook, so will never be called in this block

Reply:
The way to test this feature is to add an async promise on the page.

```vue
<script setup>
useHead({
  title: 'pre resolve'
})
await new Promise((resolve) => {
  setTimeout(() => {
    resolve()
  }, 500)
})
useHead({
  title: 'post resolve'
})
</script>
```

Then load this page via direct SSR load and also navigate to it from another page using CSR.

In theory, the `post resolve` should be the SSR title. To avoid flashing to `pre resolve` -> `post resolve` client side we need to await the Vue component, which is handled by the suspense (which may not be implemented in bridbge?).

Since the SSR suspense awaiting doesn't seem to work  (the above code snippet currently renders `pre resolve` ssr), I think we may need to skip this feature. 

This will introduce a potential flash of the title/classes for users and it may lead to unexpected ssr data, but not sure what else the solution would be.

If we did want to implement something, then unpausing the dom on idle would be the only safe route. 

```ts
requestIdleCallback(unpauseDom)
```

Reply:
Thank you for your kind review!
I remove pausing / unpausing feature.
Somehow I tried to support it in Nuxt 2, but it's difficult.

https://github.com/nuxt/bridge/pull/804/commits/96eef653e8db4110a296bc192054f4f694bd5805